### DISCUSSION TITLE
generic attribute types

### DISCUSSION THREAD

**Discussion by staabm** [rectorphp/rector#7941] [Ideas]
currently we are using `setAttribute` and `getAttribtue` on `Node` in a lot of places.

this api is not typed though.

I wonder whether we can e.g. utilize generics or create a small PHPStan extension, to make the PHPStan analysis in rector-src repo aware of the types involved when working with attributes. IMO depending on the used attribute we most of the time know which type is involved...

wdyt?

**Comment by TomasVotruba** [comment]
Sounds good :+1: 

Something like this one? 
https://github.com/rectorphp/phpstan-rules/blob/main/src/ReturnTypeExtension/GetAttributeReturnTypeExtension.php

**Reply by staabm** [reply]
ooh lol. exactly. 😎
